botorch/acquisition/cost_aware.py

```diff
@@ -201,10 +201,7 @@ def forward(
         # this will be of shape `num_fantasies x batch_shape` or `batch_shape`
         cost = cost.clamp_min(self._min_cost).sum(dim=-1)
 
-        # if we are doing inverse weighting on the sample level, clamp numerator.
-        if not self._use_mean:
-            deltas = deltas.clamp_min(0.0)
-
         # compute and return the ratio on the sample level - If `use_mean=True`
-        # this operation involves broadcasting the cost across fantasies
-        return deltas / cost
+        # this operation involves broadcasting the cost across fantasies.
+        # We multiply by the cost if the deltas are <= 0, see discussion #2914
+        return torch.where(deltas > 0, deltas / cost, deltas * cost)
```

Comment:
Thanks for putting this up! Could you update the doc-string of `forward` here, as well as the [math in the class description](https://github.com/pytorch/botorch/blob/4b8f7fcfbbf6ea8e614492226e794c6cb082b62c/botorch/acquisition/cost_aware.py#L95)? Otherwise looks good to me and ready to merge.

Reply:
Hi @SebastianAment and @esantorella 

I've rerun the MOMFBO notebook and updated the relevant doc-strings. The other 3 notebooks that use the InverseCostWeightedUtility weren't significantly affected by the change; or rather they don't use manually set seeds as with the MOMFBO notebook, so I could not see a significant difference in the results.

Thanks!